libraries/mbed/targets/hal/TARGET_ARM_SSG/TARGET_IOTSS/i2c_api.c

```diff
@@ -0,0 +1,542 @@
+                    i2c_send_ack(obj);
+
+                    // Next byte
+                    //rxdata = ((rxdata << 8) & 0xFFFFFF00);
+                    //rxdata |= (i2c_receive_byte(obj) & 0xFF);
+                    rxdata = i2c_receive_byte(obj);
+                    data[(length-1)-bytes_read] = (char)rxdata;
+                    bytes_read++;
+                            
+                }
+            }
+            break;
+        case I2C_1:
+            // Set serial and register address
+            i2c_send_byte(obj,sadr);
+            ack += i2c_receive_ack(obj);
+            i2c_send_byte(obj, address);
+            ack += i2c_receive_ack(obj);
+
+            // Stop bit
+            i2c_stop(obj);
+
+            // Start bit
+            i2c_start_tsc(obj);
+            // Fall through to read data
+        case I2C_2:
+        case I2C_3:
+            // Read from preset register address pointer
+            i2c_send_byte(obj,sadr | 1);
+            ack += i2c_receive_ack(obj);
+
+            rxdata = i2c_receive_byte(obj);
+			data[bytes_read] = (char)rxdata;
+			bytes_read++;
+            // Read multiple bytes
+            if ((length > 1) && (length < 5))
+            {
+                for (loop = 1; loop <= (length - 1); loop++)
+                {
+                    // Send ACK
+                    i2c_send_ack(obj);
+
+                    // Next byte
+                    rxdata = i2c_receive_byte(obj);
+                    data[loop] = (char)rxdata;
+                    bytes_read++;
+                            
+                }
+            }
+    		break;
+    }
+    i2c_send_nack(obj);  
+
+	i2c_stop(obj);    // Actual stop bit
+
+    return bytes_read;
+}
+
+int i2c_write(i2c_t *obj, int address, const char *data, int length, int stop)
+{
+	int ack=0;
+	int sadr;
+    char * ptr;
+    char addr;
+    ptr = (char*)data;
+    switch ((int)obj->i2c)
+    {
+        case I2C_0: 
+            sadr = TSC_I2C_ADDR;
+            addr = address;
+            break;
+        case I2C_1: 
+            sadr = AAIC_I2C_ADDR; 
+            addr = address;
+            break;
+        case I2C_2: 
+        case I2C_3: 
+            sadr = address; //LM75_I2C_ADDR or MMA7660_I2C_ADDR;
+            addr = *ptr++;
+            break;
+ 	}
+    
+//    printf("adr = %x, reg = %x\n",sadr, address);
+    i2c_start(obj);
+
+    // Set serial and register address
+    i2c_send_byte(obj,sadr);
+    ack += i2c_receive_ack(obj);
+    i2c_send_byte(obj, addr);
+    ack += i2c_receive_ack(obj);
+
+    for(int i = 1; i<length; i++)
+	{
+        i2c_send_byte(obj, *ptr++);
+        ack += i2c_receive_ack(obj);
+	}
+
+    i2c_stop(obj);
+    if(ack==3) { return 1; }
+    else{ return 0; }
+
+}
+
+void i2c_reset(i2c_t *obj) {
+    i2c_stop(obj);
+}
+
+int i2c_byte_read(i2c_t *obj, int last) {
+    return 0;
+}
+
+int i2c_byte_write(i2c_t *obj, int data) {
+    return 0;
+}
+
+void i2c_slave_mode(i2c_t *obj, int enable_slave) {
+}
```

Comment:
Slave is enabled device.h but not implemented? Please disable, and this placeholder shall be removed, you can keep it local on your stash/branch?